src/components/UserPlan.tsx

```diff
@@ -1,42 +1,207 @@
+import { Tooltip } from "./Tooltip";
+import { UpgradePlanForm } from "./UpgradePlanForm";
 
 interface UserPlanProps {
-  plan: Plan;
+  user: User;
 }
 
-export const UserPlan = ({ plan }: UserPlanProps) => {
+export const UserPlan = ({ user }: UserPlanProps) => {
+  const { sms, plan, hasClientManager, isLastPlanRequested, clientManager } =
+    user;
+
   const {
     planName,
     maxSubscribers,
+    planType,
     itemDescription,
     buttonUrl,
     buttonText,
     pendingFreeUpgrade,
     remainingCredits,
     description,
+    isSubscribers,
+    isMonthlyByEmail,
   } = plan;
 
+  const {
+    smsEnabled,
+    remainingCredits: smsRemainingCredits,
+    description: smsDescription,
+    buttonUrl: smsButtonUrl,
+    buttonText: smsButtonText,
+  } = sms;
+
+  const isPlanTypeMonthlyDeliveries = planType === "monthly-deliveries";
+  const isPlanTypeSuscribers = planType === "suscribers";
+
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const openModalHandler = () => {
+    setIsModalOpen(true);
+  };
+
+  const renderModalButton = () => (
+    <OpenModalButton className="user-plan" openModalHandler={openModalHandler}>
+      {buttonText}
+    </OpenModalButton>
+  );
+
+  const renderModalButtonWithTooltip = () => (
+    <div className="dp-request-sent">
+      <Tooltip>
+        <OpenModalButton
+          className="user-plan close-user--menu dp-tooltip-left"
+          openModalHandler={openModalHandler}
+        >
+          <FormattedMessage id="header.send_request" />
+          <div className="tooltiptext">
+            <FormattedMessage id="header.tooltip_last_plan" />
+          </div>
+        </OpenModalButton>
+        <span className="ms-icon icon-info-icon"></span>
+      </Tooltip>
+    </div>
+  );
+
+  const renderPlanLink = () => {
+    if (buttonUrl && !pendingFreeUpgrade)
+      return (
+        <a className="user-plan" href={buttonUrl}>
+          {buttonText}
+        </a>
+      );
+  };
+
   return (
     <div className="user-plan--container">
-      <div className="user-plan--type">
-        <p className="user-plan--monthly-text">
-          <strong>{planName}</strong> ({maxSubscribers} {itemDescription})
-        </p>
-        {buttonUrl && !pendingFreeUpgrade && (
-          <a className="user-plan" href={buttonUrl}>
-            {buttonText}
-          </a>
-        )}
-        <p className="user-plan--monthly-text" />
-      </div>
-
-      <div className="user-plan--type">
-        <div className="user-plan--buyContainer">
-          <p>
-            <strong>{remainingCredits}</strong> {description}
-          </p>
-        </div>
-      </div>
+      {!hasClientManager && (
+        <>
+          <UserPlanType>
+            {isSubscribers || isMonthlyByEmail ? (
+              <>
+                <MonthlyPlan>
+                  <strong>{planName}</strong> ({maxSubscribers}{" "}
+                  {itemDescription})
+                </MonthlyPlan>
+                {renderPlanLink()}
```

Comment:
Creo que si se renombra la función `renderPlanLink` por `PlanLink` se podría escribir así, no?

```suggestion
                <PlanLink />
```

Reply:
Opté por la llamada directa a la función porque las creé para organizar el código sin intención de que sean componentes porque no necesitan estado ni ciclo de vida. Y el JSX que devuelven es bien espécifico del lugar donde las estoy invocando y no son muy reutilizables

Reply:
De esta manera estamos construyendo la función en cada render y luego renderizándola, aquí yo optaría por crear un átomo pensando en Atomic Design y complementaría esta idea con un Memo.

Reply:
Acá dejo algo de código como sugerencia: https://github.com/FromDoppler/doppler-menu-mfe/pull/87/files#r904054572

Reply:
Aunque la funcion se construye cada vez, su costo es mínimo. Es apenas recrear un tag `<a>`. Igual tomé tu sugerencia e hice un profiling para comparar el tiempo de renderizarido y no veo una diferencia sgnificativa:
**Component con memo**: 1ms en el primer render y 0.3ms a partir del segundo render.
**Function**: 1.1ms en el primer render y 0.3ms a partir del segundo render.

**Component con memo**
![image](https://user-images.githubusercontent.com/94080926/175171952-d54a8f2f-0061-4ef1-b65c-7bf35b6a26c8.png)
![image](https://user-images.githubusercontent.com/94080926/175173313-9736daa2-fb2c-4437-a496-dad5207cb52e.png)


**Function**
![image](https://user-images.githubusercontent.com/94080926/175172209-55ce4341-dc02-48a7-a856-6959427da03d.png)
![image](https://user-images.githubusercontent.com/94080926/175173530-af40ebbf-2cef-402e-bfd7-55c321e483c9.png)